naga/src/proc/constant_evaluator.rs

```diff
@@ -2497,6 +2520,121 @@ impl<'a> ConstantEvaluator<'a> {
 
         Ok(resolution)
     }
+
+    fn select(
+        &mut self,
+        reject: Handle<Expression>,
+        accept: Handle<Expression>,
+        condition: Handle<Expression>,
+        span: Span,
+    ) -> Result<Handle<Expression>, ConstantEvaluatorError> {
+        let arg = |this: &mut Self, expr| {
+            let expr = this.check_and_get(expr)?;
```

Comment:
It seems to be conventional to handle the `check_and_get` calls in `try_eval_and_append_impl`, before we pass off to the helper function like this one. I don't really see why that's the convention, but let's try to stick with it for now.

Reply:
uh

Meaning, let's eliminate the `check_and_get` step here?

Reply:
Just confirmed that both `check_and_get` _and_ `try_eval_and_append` here weren't necessary; force-pushed f03f9bd65942.

Reply:
I didn't mean that `check_and_get` wasn't necessary - I think it *is*. I meant that it's conventional to call it in `try_eval_and_append_impl`, before passing off the rest of the work to an expression-specific helper function. Like:

```rust
            Expression::Access { base, index } => {
                let base = self.check_and_get(base)?;
                let index = self.check_and_get(index)?;

                self.access(base, self.constant_index(index)?, span)
            }
```

Reply:
Ah, I understand now. Re-introduced the `check_and_get` calls with a force-push to e84c6f12.